### ISSUE TITLE
Questions on the use of ground-truth labels for validation

### ISSUE THREAD

**Issue by wurenzhi** [JieyuZ2/wrench#17]
Thanks for putting up the benchmark! This is really great work!
It seems that both the label model and the end model use the ground-truth labels for validation. 
For example, the base label model uses the ground-truth labels of the validation set to calculate the class balance weights: https://github.com/JieyuZ2/wrench/blob/544119e781d010797cf153307aa1090361c99522/wrench/basemodel.py#L286
I have a few questions regarding this:
(1) A valid baseline for the label models would be a classifier trained on the validation set with the weak labels of LFs as features and the ground-truth labels as the target. Given that the validation set for most datasets is actually not small, I feel the trained model might be a pretty strong baseline compared to other unsupervised label models. 
(2) Similar to how we combine the weak labels on the training set to get aggregated labels, we could also get aggregated labels for the validation set. Then, the aggregated labels instead of the ground-truth labels of the validation set could be used for validation purposes for the end model. Wouldn't this be a more realistic setting? Especially considering that the proposal of weak supervision is to replace human labeling with programmatical labeling. 

I appreciate any explanations. Thanks!

**Comment by JieyuZ2** [comment]
Hey Renzhi,

That's a great question and thanks for pointing this out!! 

During the development of WRENCH, we do have a lot of debate about whether/how to use the validation set. But finally, we decided to follow the convention of previous work and just use a clean validation set. I agree with your arguments and currently lean to the setting wherein (1) we do not use a validation set for the label model and (2) for the end model, we use validation sets with labels produced by the label model instead of ground truth label or we can use a linear model where validation set is not necessary.

We're very happy that you found WRENCH is useful, but as you saw, WRENCH is far from perfect. We will continue working on it and make it more realistic! 

Thanks,
Jieyu

**Comment by wurenzhi** [comment]
Hi Jieyu,
Thanks a lot for your explanation! It is very helpful.
